### ISSUE TITLE
Error running plugin

### ISSUE THREAD

**Issue by alveshelio** [cypress-io/cypress#18809]
### Current behavior

I'm using Cypress with Next.js

When trying to run my tests I get this error
**Title:** Error running plugin

**Message:** The following error was thrown by a plugin. We stopped running your tests because a plugin crashed. Please check your plugins file (`/cypress/plugins/index.ts`)

**Details:** 
```
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:67:19)
    at Object.createHash (node:crypto:130:10)
    at module.exports (Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/NormalModule.js:417:16)
    at handleParseError (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/NormalModule.js:471:10)
    at /Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/NormalModule.js:503:5
    at /Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/webpack/lib/NormalModule.js:358:12
    at /Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at iterateNormalLoaders (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
    at /Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/loader-runner/lib/LoaderRunner.js:236:3
    at context.callback (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at makeSourceMapAndFinish (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/ts-loader/dist/index.js:58:5)
    at successLoader (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/ts-loader/dist/index.js:40:5)
    at Object.loader (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/ts-loader/dist/index.js:23:5)
    at LOADER_EXECUTION (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/loader-runner/lib/LoaderRunner.js:119:14)
```
**Stack trace:**
```
Error: The following error was thrown by a plugin. We stopped running your tests because a plugin crashed. Please check your plugins file (`/cypress/plugins/index.ts`)
    at Object.get (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/lib/errors.js:1035:15)
    at EventEmitter.handleError (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:189:20)
    at EventEmitter.emit (events.js:376:20)
    at ChildProcess.<anonymous> (/Library/Caches/Cypress/8.7.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:19:22)
    at ChildProcess.emit (events.js:376:20)
    at emit (internal/child_process.js:910:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)

```

### Desired behavior

I'd expect the test to run

### Test code to reproduce

This is what I have in `cypress/plugins/index.ts`
```ts
/// <reference types="cypress" />
// ***********************************************************
// This example plugins/index.ts can be used to load plugins
//
// You can change the location of this file or turn off loading
// the plugins file with the 'pluginsFile' configuration option.
//
// You can read more here:
// https://on.cypress.io/plugins-guide
// ***********************************************************

const injectNextDevServer = require('@cypress/react/plugins/next')
// This function is called when a project is opened or re-opened (e.g. due to
// the project's config changing)

/**
 * @type {Cypress.PluginConfig}
 */

module.exports = (on, config) => {
  // `on` is used to hook into various events Cypress emits
  // `config` is the resolved Cypress config
  require('@cypress/code-coverage/task')(on, config)
  if (config.testingType === 'component') {
    injectNextDevServer(on, config)
  }

  return config
}
```

### Cypress Version

cypress 8.7.0  and @cypress/react 5.10.3

**Comment by ZachJW34** [comment]
@alveshelio Could you create a reproduction of the issue?

Also, I looked around and found [this issue](https://github.com/webpack/webpack/issues/14532#issuecomment-947012063) that seems to be related. Can you see if the workaround works for your use case?

**Comment by boylec** [comment]
@alveshelio Are you using Node 17? I believe I was seeing this as well using Node 17 so I rolled back to 16.*

**Comment by alveshelio** [comment]
> @alveshelio Could you create a reproduction of the issue?
> 
> Also, I looked around and found [this issue](https://github.com/webpack/webpack/issues/14532#issuecomment-947012063) that seems to be related. Can you see if the workaround works for your use case?

Indeed it was related. I was able to use `export NODE_OPTIONS=--openssl-legacy-provider` as a work around but I guess for now I'll simply roll back to node 16* as suggested by @boylec.

Any idea of a ETA to fix this so it works well with Node 17?

Thank you very much guys for the help, much appreciated.

**Comment by ZachJW34** [comment]
The issue I linked above is Webpack related, so there is nothing on our end that we would do to fix it since we don't control your version of Webpack. If you believe the issue to be Cypress specific, then a reproduction would help me find out what's going on.

**Comment by aryzing** [comment]
How is the version of webpack set?

**Comment by Falven** [comment]
> 

You need to update the version of WebPack that Cypress uses as you are using a version that uses insecure legacy OpenSSL algorithms. Node 17+ requires this by default.
 
```
The OpenSSL legacy provider supplies OpenSSL implementations of algorithms that have been deemed legacy. Such algorithms have commonly fallen out of use, have been deemed insecure by the cryptography community, or something similar. We can consider this the retirement home of cryptographic algorithms.
```

**Comment by emilyrohrbough** [comment]
As @ZachJW34 mentioned, this issue with with Webpack & Node 17.   It looks like you are using Cypress' default file:processor and we have made a few changes to do this for you in 9.2.1+, however theres been some edge cases: (#19320, #19712, etc.). Your best option is to `NODE_OPTIONS=--openssl-legacy-provider` yourself to ensure this is pass correctly if you'd like to continue to use node 17. 

Closing in favor of https://github.com/cypress-io/cypress/issues/19712.
